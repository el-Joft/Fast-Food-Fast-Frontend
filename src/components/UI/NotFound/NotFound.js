import React from 'react';
import { Link } from 'react-router-dom';
import './NotFound.scss';


const NotFound = () => (
  <div className='notFound'>
    <Link className='link' to='/'>
      <h3 className='notfoundtext'>...go back Home...</h3>
    </Link>
  </div>
);
export default NotFound;
