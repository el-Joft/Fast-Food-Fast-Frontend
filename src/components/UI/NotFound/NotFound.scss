// 404 page
.notFound {
  background: url('/public/images/404.png') no-repeat;
  background-size: cover;
  height: 700px;
  width: 100vw;
  padding: 10px;
  margin-top: 10px;
}

.notfoundtext {
  color: #ffff;
  font-size: 30px;
  text-align: center;
  margin-top: 50px;
  border: 1px solid #ffff;
  padding: 20px;
  width: 700px;
}
.link:hover {
  color: #ffff;
}

