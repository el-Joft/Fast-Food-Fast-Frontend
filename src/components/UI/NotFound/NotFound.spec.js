import React from 'react';
import { shallow } from 'enzyme';
import { NotFound } from './NotFound';

describe('shallow testing', () => {
  let wrapper;
  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NotFound />);
    expect(wrapper.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NotFound />);
    const li = wrapper.find('div');
    expect(li.length).toEqual(1);
  });


  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NotFound />);
    const li = wrapper.find('Link');
    expect(li.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NotFound />);
    const li = wrapper.find('h3');
    expect(li.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
