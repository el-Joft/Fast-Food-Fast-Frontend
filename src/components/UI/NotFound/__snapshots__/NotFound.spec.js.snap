// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow testing should test the component if it renders anything 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="link"
        replace={false}
        to="/"
      >
        <h3
          className="notfoundtext"
        >
          ...go back Home...
        </h3>
      </Link>,
      "className": "notFound",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h3
          className="notfoundtext"
        >
          ...go back Home...
        </h3>,
        "className": "link",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "...go back Home...",
          "className": "notfoundtext",
        },
        "ref": null,
        "rendered": "...go back Home...",
        "type": "h3",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="link"
          replace={false}
          to="/"
        >
          <h3
            className="notfoundtext"
          >
            ...go back Home...
          </h3>
        </Link>,
        "className": "notFound",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3
            className="notfoundtext"
          >
            ...go back Home...
          </h3>,
          "className": "link",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "...go back Home...",
            "className": "notfoundtext",
          },
          "ref": null,
          "rendered": "...go back Home...",
          "type": "h3",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
